test/hotspot/jtreg/compiler/c2/irTests/TestVectorFPReduction.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2024, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2.irTests;
+
+import compiler.lib.ir_framework.*;
+
+/*
+ * @test
+ * @bug 8320725
+ * @summary Ensure strictly ordered AddReductionVF/VD nodes are generated on SVE machines
+ * while being disabled on Neon
+ * @requires os.arch == "aarch64"
+ * @library /test/lib /
+ * @run driver compiler.c2.irTests.TestVectorFPReduction
+ */
+
+public class TestVectorFPReduction {
+
+    final private static int SIZE = 1024;
+
+    private static double[] da = new double[SIZE];
+    private static double[] db = new double[SIZE];
+    private static float[] fa = new float[SIZE];
+    private static float[] fb = new float[SIZE];
+    private static float fresult;
+    private static double dresult;
+
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    @Test
+    @IR(applyIf = {"UseSVE", "0"}, failOn = {IRNode.ADD_REDUCTION_VF})
```

Comment:
I realize that this might fail when extended on other architectures. I will modify the rules and add `applyIfCPUFeature` instead of testing the VM flags directly (which can be modified and result in incorrect results). Will update a PS soon.